### PR TITLE
(PC-7917) App Native - Amélioration tech - Mise à jour de lingui v2 -> v3 avec traduction au pluriel

### PR DESCRIPTION
**Pull request by kopax-polyconseil**
## Jira

https://passculture.atlassian.net/browse/PC-7917

## Original PR

https://github.com/pass-culture/pass-culture-app-native/pull/842

## Things done

- [X] Inside `test/utils.tsx` extracted a render function with a wrapper with Lingui, in this way we don't need to wrap any render function on every test, also exports every function of testing library native so you could use that file for every test
- [X] Removed the `_.(t)` function because it was useless, Lingui v3 is use the macro functions, there's no need to complex everything. https://lingui.js.org/ref/macro.html#examples-of-js-macros. IN some cases you need to format currencies, numbers, or dates giving a locale, Lingui also exports date, and number macros but those ones are extracted from the @lingui/core library. **I've remember this because I saw hard-written dates formatted to iso but could be good to format them using the locale instead of the iso... **
- [X] Added `lingui.config.js` with a typedef for checking that the config is valid ;)
- [X] Modified the add new locale and add new translation readmes to match latest lingui version
- [X] Also i saw that you use a lot unicode chars to give spaces to things, in Lingui v3 we strip this characters so just use html entities or basic templating
## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [x] Added new reusable components to the AppComponents page and communicate to the team on slack

### CODE REVIEW

**Review by blixit — APPROVED** [review]
Huge ! n'oublie pas de faire un message aux PO pour qu'ils fassent attention aux traductions.

Je pense qu'il serait même préférable de demander une repasse sur les traductions lors de la prochaine session globale de test

**Comment by kopax-polyconseil** [comment]
Y a encore quelque pétouilles, en cours de fix via https://github.com/lingui/js-lingui/issues/1022

File: .eslintignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ __mocks__
 
 # files in src/api/gen are auto generated by swagger-codegen
 src/api/gen
+src/locales/**/*
```
Comment (BenStirrup) [inline .eslintignore:11]: je pense que tu peux le mettre dans le block au-dessus, pour pas faire l'amalgame avec swagger-odegen

File: src/App.test.tsx [hunk 2]
```diff
@@ -1,12 +1,11 @@
-import { render } from '@testing-library/react-native'
 import React, { useEffect } from 'react'
 import { ErrorBoundary } from 'react-error-boundary'
 import { act } from 'react-test-renderer'
 
 import { AsyncErrorBoundary } from 'features/errors'
 import { errorMonitoring } from 'libs/errorMonitoring'
 import * as BatchLocalLib from 'libs/notifications'
-import { flushAllPromises } from 'tests/utils'
+import { flushAllPromises, render } from 'tests/utils'
```
Comment (BenStirrup) [inline src/App.test.tsx:8]: import non utilisé
```suggestion
import { flushAllPromises } from 'tests/utils'
```

**Review by BenStirrup — APPROVED** [review]
Ne pas oublier dans la prochaine pr d'ajouter une règle eslint pour interdir les import depuis testing library

File: .eslintignore (hunk 1 above)
Comment (kopax-polyconseil) [reply .eslintignore:11]: OK

**Comment by kopax-polyconseil** [comment]
> Ne pas oublier dans la prochaine pr d'ajouter une règle eslint pour interdir les import depuis testing library

Après ce ticket OK

**Pull request merged**